Classes/DataProcessing/LanguageMenuProcessor.php

```diff
@@ -0,0 +1,494 @@
+     *
+     * @var array
+     */
+    protected $processorConfiguration;
+
+    /**
+     * Allowed configuration keys for menu generation, other keys
+     * will throw an exception to prevent configuration errors.
+     *
+     * @var array
+     */
+    public $allowedConfigurationKeys = [
+        'cache_period',
+        'minItems',
+        'minItems.',
+        'maxItems',
+        'maxItems.',
+        'begin',
+        'begin.',
+        'alternativeSortingField',
+        'alternativeSortingField.',
+        'excludeUidList',
+        'excludeUidList.',
+        'excludeDoktypes',
+        'alwaysActivePIDlist',
+        'alwaysActivePIDlist.',
+        'protectLvar',
+        'addQueryString',
+        'addQueryString.',
+        'if',
+        'if.',
+        'languages',
+        'languages.',
+        'as',
+        'dataProcessing',
+        'dataProcessing.'
+    ];
+
+    /**
+     * Remove keys from configuration that should not be passed
+     * to HMENU to prevent configuration errors
+     *
+     * @var array
+     */
+    public $removeConfigurationKeysForHmenu = [
+        'languages',
+        'languages.',
+        'as',
+        'dataProcessing',
+        'dataProcessing.'
+    ];
+
+    /**
+     * @var array
+     */
+    protected $menuConfig = [
+        'special' => 'language',
+        'wrap' => '[|]'
+    ];
+
+    /**
+     * @var array
+     */
+    protected $menuLevelConfig = [
+        'doNotLinkIt' => '1',
+        'wrapItemAndSub' => '{|}, |*| {|}, |*| {|}',
+        'stdWrap.' => [
+            'cObject' => 'COA',
+            'cObject.' => [
+                '10' => 'LOAD_REGISTER',
+                '10.' => [
+                    'languageUid.' => [
+                        'cObject' => 'TEXT',
+                        'cObject.' => [
+                            'value' => '', // Will be set by prepareConfiguration
+                            'listNum.' => [
+                                'stdWrap.' => [
+                                    'data' => 'register:count_HMENU_MENUOBJ',
+                                    'wrap' => '|-1'
+                                ],
+                                'splitChar' => ','
+                            ]
+                        ]
+                    ]
+                ],
+                '20' => 'USER',
+                '20.' => [
+                    'userFunc' => 'TYPO3\CMS\Frontend\DataProcessing\LanguageMenuProcessor->getFieldAsJson',
+                    'language.' => [
+                        'data' => 'register:languageUid'
+                    ],
+                    'field' => 'title',
+                    'stdWrap.' => [
+                        'wrap' => '"title":|'
+                    ]
+                ],
+                '21' => 'TEXT',
+                '21.' => [
+                    'value' => self::LINK_PLACEHOLDER,
+                    'wrap' => ',"link":|',
+                ],
+                '22' => 'TEXT',
```

Comment:
do we really need a target here?

Reply:
for me it's not really useful but what if hmenu delivers it?

Reply:
we can just drop it, it hmenu would deliver it - the current page would be set to blank for exmaple, that would result in switching the language would open a new tab - i think we can savely drop it.